Source/Utility/Diagnostics/DiagProbe.cpp

```diff
@@ -0,0 +1,324 @@
+#include "DiagProbe.H"
+#include <AMReX_VisMF.H>
+#include <AMReX_FPC.H>
+#include <AMReX_PlotFileUtil.H>
+#include <regex>
+#include <cstdio>
+
+AMREX_GPU_DEVICE
+AMREX_FORCE_INLINE
+amrex::Real
+LinearInterpolate(
+  amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> xp,
+  amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> x_low,
+  amrex::Array3D<amrex::Real, 0, 1, 0, 1, 0, 1> cell_data,
```

Comment:
Change the variable name from `cell_data`, which might lead to confusion because there is a different AMReX data structure called cellData. But also, instead of creating the new cell_data structure and copying data into it, why not just pass the original array4 and the relevant index in?

Reply:
Renamed the variable name.
Do you think changing it to array4 can get some benefits? If not, I would like to keep it this way and avoid rewriting the code once again

Reply:
Sure, maybe the code could be a bit simpler by not creating the intermediate data structure, but if what's there works on GPU it should be fine and not worth rewriting.